tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/TestFunctions.py

```diff
@@ -166,3 +166,176 @@ def alfven_dt_magnetic_field(x, t, wavenumber, pressure, rest_mass_density,
+      (n_s_c_2 / (g - 1.0) * (2.0 + n_s_c_2 / (g - 1.0)))
+
+
+def bernoulli_equation_lhs_squared_minus_one(rho, r, g, k, m_dot_over_four_pi, mass):
+    u = m_dot_over_four_pi / (r**2 * rho)
+    g_minus_one = g - 1.0
+    #polytropic_index times newtonian_sound_speed_squared:
+    ncns2 = k * g * pow(rho, g_minus_one) / g_minus_one
+    #specific enthalpy squared minus one:
+    h2_minus_one = ncns2 * (2.0 + ncns2)
+    h2 = h2_minus_one + 1.0
+    return h2_minus_one + h2 * (-2.0 * mass / r + u**2)
+
+def bernoulli_root_function(rho, r, g, k, m_dot_over_four_pi, n_s_c_2, u_c_2, mass):
+    return bernoulli_equation_lhs_squared_minus_one(rho, r, g, k, m_dot_over_four_pi, mass) \
+      - bernoulli_constant_squared_minus_one(n_s_c_2, u_c_2, g)
+
+def sound_speed_at_infinity_squared(g, c_s_c_2):
+    g_minus_one = g - 1.0
+    return g_minus_one + (c_s_c_2 - g_minus_one) * np.sqrt(1.0 + 3.0 * c_s_c_2)
+
+
+def rest_mass_density_at_infinity(g, rho_c, c_s_inf_2, c_s_c_2):
+    return rho_c * pow(c_s_inf_2/(c_s_c_2 * np.sqrt(1.0 + 3.0 * c_s_c_2)), 1.0/(g-1.0))
+
+def rest_mass_density_at_horizon(g, rho_c, u_c_2):
+    return 0.0625 * rho_c * pow(u_c_2, -1.5)
+
+
+def one_over_lapse(mass, radius):
+    return np.sqrt(1.0 + 2.0 * mass / radius)
+
+
+def shift(mass, radius):
+    return 2.0 * mass / (radius + 2.0 * mass)
+
+
+def fluid_four_velocity_u_t(mass, radius, abs_u_r):
+    if radius == 2.0 * mass:
+       return abs_u_r + 0.5 / abs_u_r
+    return (-2.0 * mass * abs_u_r + radius * \
+           np.sqrt(abs_u_r**2 + 1.0 - 2.0 * mass / radius)) / \
+           (radius - 2.0 * mass)
+
+
+def bondi_michel_mhd_variables(mass, sonic_radius, sonic_density, adiabatic_exponent):
+    u_c_2 = sonic_fluid_speed_squared(sonic_radius, mass)
+    c_s_c_2 = sonic_sound_speed_squared(u_c_2)
+    n_s_c_2 = sonic_newtonian_sound_speed_squared(adiabatic_exponent, c_s_c_2)
+    k = adiabatic_constant(n_s_c_2, adiabatic_exponent, sonic_density)
+    m_dot_over_four_pi = mass_accretion_rate_over_four_pi(sonic_radius, sonic_density, u_c_2)
+    h_inf_2 = bernoulli_constant_squared_minus_one(n_s_c_2, u_c_2, adiabatic_exponent) + 1.0
+    c_s_inf_2 = sound_speed_at_infinity_squared(adiabatic_exponent, c_s_c_2)
+    rho_inf = rest_mass_density_at_infinity(adiabatic_exponent, sonic_density, c_s_inf_2, c_s_c_2)
+    rho_horizon = rest_mass_density_at_horizon(adiabatic_exponent, sonic_density, u_c_2)
+    return {"sonic_fluid_speed_squared": u_c_2, \
+            "sonic_sound_speed_squared": c_s_c_2, \
+            "sonic_newtonian_sound_speed_squared": n_s_c_2, \
+            "adiabatic_constant": k,\
+            "mass_accretion_rate_over_four_pi": m_dot_over_four_pi, \
+            "bernoulli_constant_squared": h_inf_2, \
+            "sound_speed_at_infinity_squared": c_s_inf_2, \
+            "rest_mass_density_at_infinity": rho_inf, \
+            "rest_mass_density_at_horizon": rho_horizon}
+
+
+def bondi_michel_mhd_rest_mass_density(x, mass, sonic_radius,
```

Comment:
When you remove `MHD` from the class name don't forget to remove it in these function names too.